pkg/envoy/envoy_test.go

```diff
@@ -131,8 +137,68 @@ func (s *EnvoySuite) TestEnvoy(c *C) {
 
 	// Remove listener3 again, and wait for timeout after stopping Envoy.
 	log.Debug("removing listener 3")
-	xdsServer.RemoveListener("listener3", s.waitGroup)
+	xdsServer.RemoveListener("listener3", s.waitGroup.AddCompletion())
 	err = s.waitForProxyCompletion()
 	c.Assert(err, NotNil)
 	log.Debugf("failed to remove listener 3: %s", err)
 }
+
+func (s *EnvoySuite) TestEnvoyNACK(c *C) {
+	log.Logger.SetLevel(logrus.DebugLevel)
+	ctx, cancel := context.WithTimeout(context.Background(), 50*time.Second)
+	defer cancel()
+
+	s.waitGroup = completion.NewWaitGroup(ctx)
+
+	if os.Getenv("CILIUM_ENABLE_ENVOY_UNIT_TEST") == "" {
+		c.Skip("skipping envoy unit test; CILIUM_ENABLE_ENVOY_UNIT_TEST not set")
+	}
+
+	flowdebug.Enable()
+
+	stateLogDir, err := ioutil.TempDir("", "envoy_go_test")
+	c.Assert(err, IsNil)
+
+	log.Debugf("state log directory: %s", stateLogDir)
+
+	xdsServer := StartXDSServer(stateLogDir)
+	defer xdsServer.stop()
+	StartAccessLogServer(stateLogDir, xdsServer, &dummyEndpointInfoRegistry{})
+
+	// launch debug variant of the Envoy proxy
+	envoyProxy := StartEnvoy(stateLogDir, filepath.Join(stateLogDir, "cilium-envoy.log"), 42)
+	c.Assert(envoyProxy, NotNil)
+	log.Debug("started Envoy")
+
+	ProxyPort := uint16(80) // Port 80 requires privileges we dont have when running this test
```

Comment:
That's not true. Unit tests are run with `sudo`, cf. `Makefile`:
```
unit-tests: start-kvstores
    ...
    sudo go test ...
```
You should pick a port that should always be already open, e.g. port `22` (ssh).

Reply:
Did not face this as the tests requiring installed `cilium-envoy`are disabled by default. Changing the port to 22, will test sudoed as well.